### PR TITLE
Increase warning level and enable warning as error

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

File: CMakeSettings.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
       "inheritEnvironments": [ "msvc_x64_x64" ],
       "buildRoot": "${projectDir}\\out\\build\\${name}",
       "installRoot": "${projectDir}\\out\\install\\${name}",
-      "cmakeCommandArgs": "-DBUILD_SAMPLES=true -DUSE_CPPRESTSDK=true",
+      "cmakeCommandArgs": "-DBUILD_SAMPLES=true -DUSE_CPPRESTSDK=true -DWERROR=true -DWALL=true",
```
Comment (anurse) [inline CMakeSettings.json:10]: Is this needed when the CMakeLists already turns it on? Seems redundant unless I'm misreading the behavior.

File: src/signalrclient/CMakeLists.txt [hunk 2]
```diff
@@ -31,6 +31,40 @@ include_directories(
...
+    target_compile_options(signalrclient PRIVATE
+        /wd4820 # padding added after data member
+        /wd4514 # unreferenced inline function removed
+        /wd5045 # compiler will insert Spectre mitigation if /Qspectre switch is added
+        /wd4464 # relative include paths
+        /wd4711 # function 'x' selected for automatic inline expansion
+        /wd4710 # function not inlined
+        /experimental:external /external:anglebrackets /external:templates- /external:W0
+        /GR- # disable run-time type information
+        /guard:cf # enable control-flow guard
```
Comment (BrennanConroy) [inline src/signalrclient/CMakeLists.txt:50]: Making @blowdart happier 😄

File: CMakeSettings.json (hunk 1 above)
Comment (BrennanConroy) [reply CMakeSettings.json:10]: Ah true

File: include/signalrclient/signalr_value.h [hunk 3]
```diff
@@ -180,8 +180,20 @@ namespace signalr
             double number;
             std::map<std::string, value> map;
 
+            // constructor is not implicitly called
+#pragma warning (push)
+#pragma warning (disable: 4582)
```
Comment (halter73) [inline include/signalrclient/signalr_value.h:185]: Has this false-positive been fixed? https://developercommunity.visualstudio.com/content/problem/564332/stdoptional-generates-warning-c4582-constructor-is.html

Also, do these pragmas only apply to msvc?

Comment (BrennanConroy) [reply include/signalrclient/signalr_value.h:185]: It's not a false positive.

> Also, do these pragmas only apply to msvc?

Yes

**Pull request merged**